### PR TITLE
Feature/replace deprecated finders

### PR DESCRIPTION
**Pull request by NobodysNightmare**
## OpenProject work package

https://community.openproject.org/work_packages/20325

### CODE REVIEW

File: lib/open_project/costs/patches/work_packages_helper_patch.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def work_package_form_all_middle_attributes_with_costs(form, work_package, local
 
       def work_package_form_budget_attribute(form, _work_package, _locals)
         field = work_package_form_field {
-          options = CostObject.find_all_by_project_id(@project, order: 'subject ASC').map { |d| [d.subject, d.id] }
+          options = CostObject.find_all_by(project_id: @project, order: 'subject ASC').map { |d| [d.subject, d.id] }
```
Comment (myabc) [inline lib/open_project/costs/patches/work_packages_helper_patch.rb]: `.where`?

File: app/models/cost_type.rb [hunk 2]
```diff
@@ -33,9 +33,7 @@ class CostType < ActiveRecord::Base
 
   # finds the default CostType
   def self.default
-    result = CostType.find(:first, conditions: { default: true })
-    result ||= CostType.find(:first)
-    result
+    CostType.find_by(default: true) || CostType.first
```
Comment (myabc) [inline app/models/cost_type.rb:36]: Neither of these methods should raise `ActiveRecord::RecordNotFound`

File: lib/open_project/costs/patches/work_packages_helper_patch.rb (hunk 1 above)
Comment (NobodysNightmare) [inline lib/open_project/costs/patches/work_packages_helper_patch.rb]: of course -.-

thank you ^^

File: app/models/cost_type.rb (hunk 2 above)
Comment (NobodysNightmare) [inline app/models/cost_type.rb:36]: That makes the refactoring even better ^^

File: app/models/hourly_rate.rb [hunk 3]
```diff
@@ -58,15 +58,17 @@ def self.at_date_for_user_in_project(date, user_id, project = nil, include_defau
...
-        rate = find(:first,
-                    conditions: ['user_id = ? and project_id in (?) and valid_from <= ?', user_id, project.ancestors.to_a, date],
-                    include: :project,
-                    order: 'projects.lft DESC, valid_from DESC')
+        rate = where(['user_id = ? and project_id in (?) and valid_from <= ?',
+                      user_id,
+                      project.ancestors.to_a,
+                      date])
+               .includes(:project)
+               .order('projects.lft DESC, valid_from DESC')
```
Comment (myabc) [inline app/models/hourly_rate.rb:71]: @NobodysNightmare you missed the call to `.first` here. Corrected in 5cc2403100a6c3d53be04e940314556aa50bc9b0

**Pull request merged**